commands/create_cluster.go

```diff
@@ -245,11 +250,11 @@ func validateCreateClusterPreConditions(args addClusterArguments) error {
 
 	// false flag combination?
 	if args.inputYAMLFile != "" {
-		if args.numWorkers != 0 || args.workerNumCPUs != 0 || args.workerMemorySizeGB != 0 || args.workerStorageSizeGB != 0 {
+		if args.numWorkers != 0 || args.workerNumCPUs != 0 || args.workerMemorySizeGB != 0 || args.workerStorageSizeGB != 0 || args.wokerAwsEc2InstanceType != "" {
 			return microerror.Mask(conflictingFlagsError)
 		}
 	} else {
-		if args.numWorkers == 0 && (args.workerNumCPUs != 0 || args.workerMemorySizeGB != 0 || args.workerStorageSizeGB != 0) {
+		if args.numWorkers == 0 && (args.workerNumCPUs != 0 || args.workerMemorySizeGB != 0 || args.workerStorageSizeGB != 0 || args.wokerAwsEc2InstanceType != "") {
```

Comment:
I'm a bit suspicious about this line. have you tested `gsctl` already with this new flag? I haven't taken the time to build it and try it yet. But this makes me think we'll always get a 'numWorkerNodesMissingError' when using the new instance type flag. Or does `args.numWorkers` get set by default somewhere before these preconditions?

Reply:
I have tested, yes. In fact the `--aws-instane-type` can only be used if also the `--num-workers` flag is set. Otherwise we can't generate an appropriate cluster definition to submit. We need `n` worker items, and each holds the aws.instance_type attribute. Same is the case for the CPU, RAM, and storage flags.

Reply:
So the default behaviour of 3 nodes is not a thing when we're doing aws-instance-type?

Reply:
Right. With the way our API is designed, we can't express "I want the default amount of workers, and they have to be this non-default instance type".